src/lib.rs

```diff
@@ -893,3 +895,91 @@ pub async fn build_recipes(
 
     Ok(())
 }
+
+/// Debug a recipe by setting up the environment without running the build script
+pub async fn debug_recipe(
+    debug_data: DebugData,
+    log_handler: &Option<LoggingOutputHandler>,
+) -> miette::Result<()> {
+    let recipe_path = get_recipe_path(&debug_data.recipe_path)?;
+
+    let build_data = BuildData {
+        build_platform: debug_data.build_platform,
+        target_platform: debug_data.target_platform,
+        host_platform: debug_data.host_platform,
+        channels: debug_data.channels,
+        common: debug_data.common,
+        keep_build: true,
+        debug: true,
+        test: TestStrategy::Skip,
+        up_to: None,
+        variant_config: Vec::new(),
+        ignore_recipe_variants: false,
+        render_only: false,
+        with_solve: true,
+        no_build_id: false,
+        package_format: PackageFormatAndCompression {
+            archive_type: ArchiveType::Conda,
+            compression_level: CompressionLevel::Default,
+        },
+        compression_threads: None,
+        io_concurrency_limit: num_cpus::get(),
+        no_include_recipe: false,
+        color_build_log: true,
+        tui: false,
+        skip_existing: SkipExisting::None,
+        noarch_build_platform: None,
+        extra_meta: None,
+        sandbox_configuration: None,
+    };
+
+    let tool_config = get_tool_config(&build_data, log_handler)?;
+
+    let outputs = get_build_output(&build_data, &recipe_path, &tool_config).await?;
+
+    tracing::info!("Build and/or host environments created for debugging.");
+
+    for output in outputs {
+        output
+            .build_configuration
+            .directories
+            .recreate_directories()
+            .into_diagnostic()?;
+        let output = output.fetch_sources(&tool_config).await.into_diagnostic()?;
+        let output = output
+            .resolve_dependencies(&tool_config)
+            .await
+            .into_diagnostic()?;
+        output
+            .install_environments(&tool_config)
+            .await
+            .into_diagnostic()?;
+
+        if let Some(deps) = &output.finalized_dependencies {
+            if deps.build.is_some() {
+                tracing::info!(
+                    "\nBuild dependencies available in {}",
+                    output
+                        .build_configuration
+                        .directories
+                        .build_prefix
+                        .display()
+                );
+            }
+            if deps.host.is_some() {
+                tracing::info!(
+                    "Host dependencies available in {}",
+                    output.build_configuration.directories.host_prefix.display()
+                );
+            }
+        }
+
+        tracing::info!("\nTo run the actual build, use:");
```

Comment:
Here we should print the instructions to find the "WORK_DIR" inside the environment, so that the user can `cd` there and run `bash conda_build.sh` etc. to debug.